src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -1138,6 +1135,83 @@ def add_units(
         kwargs = {key: val[i] for key, val in aggregated_unit_properties.items()}
 
         units_table.add_unit(id=int(unit_id), spike_times=spkt, **kwargs)
+
+    if write_as == "units":
+        if nwbfile.units is None:
+            nwbfile.units = units_table
+        else:
+            warnings.warn("The nwbfile already contains units. These units will not be over-written.")
+    elif write_as == "processing":
+        ecephys_mod = get_module(
+            nwbfile=nwbfile,
+            name="ecephys",
+            description="Intermediate data from extracellular electrophysiology recordings, e.g., LFP.",
+        )
+        ecephys_mod.add(units_table)
+
+
+def add_wave_forms_to_units_table(
+    sorting: SpikeInterfaceSorting,
+    nwbfile: pynwb.NWBFile,
+    skip_features: Optional[List[str]] = None,
+    units_table_name: str = "units",
+    unit_table_description: str = "Autogenerated by nwb_conversion_tools.",
+    write_units_table_in_processing_module: bool = False,
+):
+    """
+    Primary method for writing a SortingExtractor object to an NWBFile.
+    Parameters
+    ----------
+    sorting: SpikeInterfaceSorting
+    save_path: PathType
+        Required if an nwbfile is not passed. The location where the NWBFile either exists, or will be written.
+    overwrite: bool
+        If using save_path, whether or not to overwrite the NWBFile if it already exists.
+    nwbfile: NWBFile
+        Required if a save_path is not specified. If passed, this function
+        will fill the relevant fields within the nwbfile. E.g., calling
+        spikeextractors.NwbRecordingExtractor.write_recording(
+            my_recording_extractor, my_nwbfile
+        )
+        will result in the appropriate changes to the my_nwbfile object.
+    skip_features: list of str
+        Each string in this list that matches a spike feature will not be written to the NWBFile.
+    units_table_name : str (optional, defaults to 'units')
+        The name of the units table. If write_as=='units', then units_table_name must also be 'units'.
+    unit_table_description : str (optional)
+        Text description of the units table; it is recommended to include information such as the sorting method,
+        curation steps, etc.
+    write_units_table_in_processing_module: bool (optional, defaults to False)
+        How to save the units table in the nwb file.
+        - True will save it to the processing module to serve as a historical provenance for the official table.
+        - False will save it to the official NWBFile.Units position; recommended only for the final form of the data.
+    """
+    if isinstance(sorting, SortingExtractor):
+        sampling_frequency = sorting.get_sampling_frequency()
+        if sampling_frequency is None:
+            raise ValueError("Writing a SortingExtractor to an NWBFile requires a known sampling frequency!")
+        checked_sorting = OldToNewSorting(oldapi_sorting_extractor=sorting)
```

Comment:
Wait a sec, won't doing this here prevent us from entering 1215 below? In which case, doesn't this make it impossible to write waveforms that are stored on a spikeextractor object as features?

Reply:
No, the check that you are talking about is done in the original sorting extractor and not in the checked_recording:
```python
if isinstance(sorting, SortingExtractor):
```

In fact, I am removing the checked_extractor logic here. At the moment, it does not make sense.

Reply:
Aha, yes my bad. Thanks for clarifying.

Reply:
Yes, that confused me the first time btw.